sigmf/sigmffile.py

```diff
@@ -277,8 +277,8 @@ def _count_samples(self):
             sample_size = self.get_sample_size()
             num_channels = self.get_num_channels()
             sample_count = file_size // sample_size // num_channels
-            if file_size % sample_size != 0:
-                warnings.warn("File '{}' does not contain an integral number of sample. It might not be valid data.".format(self.data_file))
+            if file_size % sample_count != 0:
```

Comment:
This seems wrong. `file_size` and `sample_size` are both in bytes, so `file_size % sample_size == 0` iff the file contains a whole number of samples (single channel), but `file_size % sample_count` doesn't work the same way. As a simple example, consider what happens when `file_size == sample_size + 1` (for `sample_size > 1`).

To generalize for multiple channels, I think the condition should just be `file_size % (sample_size * num_channels) != 0`.